core/src/main/java/org/apache/calcite/sql/SqlBinaryOperator.java

```diff
@@ -186,6 +186,10 @@ public RelDataType deriveType(
       final SqlMonotonicity mono0 = call.getOperandMonotonicity(0);
       final SqlMonotonicity mono1 = call.getOperandMonotonicity(1);
       if (mono1 == SqlMonotonicity.CONSTANT) {
+        // e.g., mono / null --> constant (null)
+        if (call.isOperandNull(1, true)) {
+          return SqlMonotonicity.CONSTANT;
+        }
```

Comment:
I'm not sure if we can make the assumption "if operand1 is null the the monotonicity is always CONSTANT" for every binary operator, and i think this change is radical.

Reply:
Current runtime processes most SqlBinaryOperators with `NullPolicy.STRICT`(i.e., returns null when one of its operand is null). Should we enumerates all cases here?

Reply:
The monotonicity inference is in the "logic" layer, we should not rely on the runtime execution to ensure the semantics, enumerate the operators would be more clear to me.